Assets/Scripts/Combat/Abilities/Confection.cs

```diff
@@ -0,0 +1,277 @@
+            if(BrewCanvasGroup != null)
+                EnableCanvas(BrewCanvas, false); //this disables both the canvas and canvasgroup
+            else
+                Debug.Log("BrewCanvas CanvasGroup is Null"); 
+        }
+        else 
+        { 
+            Debug.Log("BrewCanvas is Null in Start()"); 
+        }
+
+        // Handle Bake canvas and canvasgroup
+        if(BakeCanvas != null) 
+        { 
+            // get script
+            SliderScript = BakeCanvas.GetComponent<SliderHandle>();
+            MaxClicks = SliderScript.GetNumberOfClicks();
+
+            BakeCanvasGroup = BakeCanvas.GetComponent<CanvasGroup>();
+            if(BakeCanvasGroup != null)
+                EnableCanvas(BakeCanvas, false); //this disables both the canvas and canvasgroup
+            else
+                Debug.Log("BakeCanvas CanvasGroup is Null"); 
+        }
+        else 
+        { 
+            Debug.Log("BakeCanvas is Null in Start()"); 
+        }
+    }
+
+    private void EnableCanvas(Canvas canvas, bool enabled)
+    {
+        canvas.enabled = enabled;
+        EnableCanvasGroup(canvas, enabled);
+    }
+
+    private void EnableCanvasGroup(Canvas canvas, bool enabled)
+    {
+        if(enabled)
+        {
+            if(canvas.name == "BrewCanvas")
+                BrewCanvasGroup.alpha = VisibleAlpha;
+            else if(canvas.name == "BakeCanvas")
+            {
+                SliderScript.enabled = true;
+                BakeCanvasGroup.alpha = VisibleAlpha;
+            }
+        }
+        else
+        {
+            if(canvas.name == "BrewCanvas")
+            {
+                BrewCanvasGroup.alpha = InvisibleAlpha;
+            }
+            else if(canvas.name == "BakeCanvas")
+            {
+                SliderScript.enabled = false;
+                BakeCanvasGroup.alpha = InvisibleAlpha;
+            }
+        }
+    }
+
+    new public void StartAbility()
+    {
+        base.StartAbility();
+        Debug.Log("Started Confection Ability");
+        EnableCanvas(BrewCanvas, true);
+        CurrentDamage = 0;
+    }
+
+    private void Update()
+    {
+        if(phase == Phase.Brew)
+            BrewUpdate();
+
+        if (phase == Phase.Bake)
+            BakeUpdate();
+    }
+
+    private void BrewUpdate()
+    {
+        if (timer.IsInProgress())
+        {
+            float progress = timer.GetProgress();
+
+            SweetsDropped = CookingPot.GetComponent<ItemSlot>().GetSweetsDropped();
+            RotsDropped = CookingPot.GetComponent<ItemSlot>().GetRotsDropped();
+        }
+        else
+        {
+            EndBrewPhase();
+        }
+    }
+    
+    private void BakeUpdate()
+    {
+        if (timer.IsInProgress())
+        {
+            GoodClicks = SliderScript.GetGoodClicks();
+            PerfectClicks = SliderScript.GetPerfectClicks();
+            NumberOfClicks = SliderScript.GetNumberOfClicks();
+            if(NumberOfClicks == 3)
```

Comment:
3 can be put as a variable (can be named MaxSliderClicks?). Try to avoid hard coded values as much as possible.